src/Utilities/CssDocument.php

```diff
@@ -110,175 +113,62 @@ public function getStyleRulesData(array $allowedMediaTypes): array
-     * Each part is an array where:
-     *
-     * - key "css" will contain clean CSS code (for @media rules this will be the group rule body within "{...}")
-     * - key "media" will contain "@media " followed by the media query list, for all allowed media queries,
-     *   or an empty string for CSS not within a media query
-     *
-     * Example:
-     *
-     * The CSS code
-     *
-     *   "@import "file.css"; h1 { color:red; } @media { h1 {}} @media tv { h1 {}}"
-     *
-     * will be parsed into the following array:
-     *
-     *   0 => [
-     *     "css" => "h1 { color:red; }",
-     *     "media" => ""
-     *   ],
-     *   1 => [
-     *     "css" => " h1 {}",
-     *     "media" => "@media "
-     *   ]
-     *
-     * @param array<array-key, string> $allowedMediaTypes
-     *
-     * @return array<int, array{css: string, media: string}>
-     */
-    private function splitCssAndMediaQuery(array $allowedMediaTypes): array
-    {
-        $mediaTypesExpression = '';
-        if (!empty($allowedMediaTypes)) {
-            $escapedAllowedMediaTypes = \array_map(
-                static function (string $mediaType): string {
-                    return \preg_quote($mediaType, '#');
-                },
-                $allowedMediaTypes
-            );
-            $mediaTypesExpression = '|' . \implode('|', $escapedAllowedMediaTypes);
+        if ($rule instanceof CssImport) {
+            return $this->isImportRuleAllowed;
         }
 
-        $cssSplitForAllowedMediaTypes = \preg_split(
-            '#(@media\\s++(?:only\\s++)?+(?:(?=[{(])' . $mediaTypesExpression . ')' . self::MEDIA_RULE_BODY_MATCHER
-            . ')#misU',
-            $this->styleRules,
-            -1,
-            PREG_SPLIT_DELIM_CAPTURE
-        );
+        $this->isImportRuleAllowed = false;
 
-        // filter the CSS outside/between allowed @media rules
-        $cssCleaningMatchers = [
-            'import/charset directives' => '/\\s*+@(?:import|charset)\\s[^;]++;/i',
-            'remaining media enclosures' => '/\\s*+@media\\s' . self::MEDIA_RULE_BODY_MATCHER . '/isU',
-        ];
+        if (!$rule instanceof CssAtRule) {
+            return false;
+        }
 
-        $splitCss = [];
-        foreach ($cssSplitForAllowedMediaTypes as $index => $cssPart) {
-            $isMediaRule = $index % 2 !== 0;
-            if ($isMediaRule) {
-                \preg_match('/^([^{]*+){(.*)}[^}]*+$/s', $cssPart, $matches);
-                $splitCss[] = ['css' => $matches[2], 'media' => $matches[1]];
-            } else {
-                $cleanedCss = \trim(\preg_replace($cssCleaningMatchers, '', $cssPart));
-                if ($cleanedCss !== '') {
-                    $splitCss[] = ['css' => $cleanedCss, 'media' => ''];
-                }
-            }
+        switch ($rule->atRuleName()) {
+            case 'media':
+                return false;
+            case 'font-face':
+                return $rule instanceof CssRuleSet
```

Comment:
Would a decicated method in sabberworm/php-css-parser make sense for this (i.e., should we open a ticket)?

Reply:
Ditto for other `instanceof`.  I don't think so.  I think it's perfectly succinct as it is.